Clients/GUI/Pages/UserInfo.razor

```diff
@@ -0,0 +1,133 @@
+        <br />
+        <ValidationSummary />
+        <h4 style="display:@(ErrorOccured ? "block" : "hidden"); color:red">@ErrorMessage</h4>
+        <input type="button" @onclick="@EditInfoClickHandler" class="form-control col-3 button btn-primary" value="@(IsEditModeDisabled ? "Edit" : "Save changes")" />
+    </EditForm>
+</div>
+
+@code {
+    public bool IsEditModeDisabled { get; set; }
+    public bool IsPasswordBeingChanged { get; set; }
+    public bool PopupVisible { get; set; }
+    public string ErrorMessage { get; set; }
+    public bool ErrorOccured { get; set; }
+    private UserInfoViewModel user;
+    private Guid currentId;
+
+    private void EditInfoClickHandler()
+    {
+        if (IsEditModeDisabled)
+        {
+            IsEditModeDisabled = false;
+        }
+        else // Trying to save changes
+        {
+            // Edit is inprocess in other team
+            try
+            {
+                var userInfo = new DTO.UserInformation(user.Name, user.Surname, user.DateOfBirth);
+
+                // This is in process (user/edit endpoint is another team's task)
+                if (IsPasswordBeingChanged)
+                {
+                    //UserEditor.EditUser(currentId, userInfo, user.OldPassword, user.NewPassword);
+                }
+                else
+                {
+                    //UserEditor.EditUser(currentId, userInfo, user.OldPassword, user.NewPassword);
+                }
+            }
+            // This must be changed to a custom exception later
+            catch (Exception)
+            {
+                // This will be changed to real error message
+                ErrorMessage = "error";
+                ErrorOccured = true;
+            }
+
+            Modal.Show<Modals.ChangesSavedModal>("Changes saved");
+            IsEditModeDisabled = true;
+            IsPasswordBeingChanged = false;
+        }
+    }
+
+    private void ChangePasswordClickHandler()
+    {
+        IsPasswordBeingChanged ^= true;
+    }
+
+    private void UpdateUserModel()
+    {
+        user = new UserInfoViewModel(); // TODO: get it by id (another team task)
+        user.Name = null; // to call pop-up
+        user.Surname = null;
+        // some dummy data
+        user.DateOfBirth = DateTime.Today;
+        user.Email = "john.smith@gmail.com";
+    }
+
+    protected override void OnInitialized()
+    {
+        ErrorOccured = false;
+        ErrorMessage = "";
+        PopupVisible = false;
+        IsEditModeDisabled = true;
+        IsPasswordBeingChanged = false;
+        currentId = new Guid(); // TODO: grab it from local storage
+        UpdateUserModel();
+    }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (string.IsNullOrEmpty(user.Name))
```

Comment:
Sorry, I don't get it: why do we have to wait if name is null?

Reply:
It means that user does not have any data set. It happens with the first login after registration

Reply:
And it added data that user wrote in registration form?

Reply:
> And it added data that user wrote in registration form?

If user's record in database does not contain their name, etc., then we have to ask user to fill his personal data. After adding data in modal form, the data is sent to database, so next time popup won't show.
However, not sure if checking this everytime openning this page is a good idea, so you can suggest brand new solution =)

Reply:
Now I don't have any suggestions :(